MatrixSDK/VoIP/MXCallManager.m

```diff
@@ -1732,65 +1732,80 @@ - (void)directCallableRoomWithVirtualUser:(NSString * _Nonnull)userId
 
 #pragma mark - Recent
 
-- (NSArray<MXUser *> * _Nonnull)getRecentCalledUsers:(NSUInteger)maxNumberOfUsers
-                                      ignoredUserIds:(NSArray<NSString*> * _Nullable)ignoredUserIds
+- (void)getRecentCalledUsers:(NSUInteger)maxNumberOfUsers
+              ignoredUserIds:(NSArray<NSString*> * _Nullable)ignoredUserIds
+                  completion:(nonnull void (^)(NSArray<MXUser *> * _Nonnull))completion
 {
     if (maxNumberOfUsers == 0)
     {
-        return NSArray.array;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completion(@[]);
+        });
+        return;
     }
     
     NSArray<MXRoom *> *rooms = _mxSession.rooms;
     
     if (rooms.count == 0)
     {
-        return NSArray.array;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completion(@[]);
+        });
+        return;
     }
     
     NSMutableArray *callEvents = [NSMutableArray arrayWithCapacity:rooms.count];
     
+    dispatch_group_t dispatchGroup = dispatch_group_create();
     for (MXRoom *room in rooms) {
-        id<MXEventsEnumerator> enumerator = [room enumeratorForStoredMessagesWithTypeIn:@[kMXEventTypeStringCallInvite]];
-        MXEvent *callEvent = enumerator.nextEvent;
-        if (callEvent)
-        {
-            [callEvents addObject:callEvent];
-        }
+        dispatch_group_enter(dispatchGroup);
+        [room enumeratorForStoredMessagesWithTypeIn:@[kMXEventTypeStringCallInvite]
```

Comment:
I can see now a bug on this part of the code. This is not due to the changes to this PR.
`enumeratorForStoredMessagesWithTypeIn` returns only raw content of events. It does not decrypt them. It cannot see message type.